test/datasets/commonvoice_test.py

```diff
@@ -0,0 +1,80 @@
+import os
+import csv
+import random
+
+from torchaudio.datasets import commonvoice
+from ..common_utils import (
+    TempDirMixin,
+    TorchaudioTestCase,
+    get_whitenoise,
+    save_wav,
+    normalize_wav,
+)
+
+
+class TestCommonVoice(TempDirMixin, TorchaudioTestCase):
+    backend = 'default'
+
+    root_dir = None
+    data = []
+    _headers = [u"client_ids", u"path", u"sentence", u"up_votes", u"down_votes", u"age", u"gender", u"accent"]
+    _client_ids = [u"0", u"1"]
+    _sentence = [u"test", u"pytorch"]
+    _up_votes = [u"0", u"1"]
+    _down_votes = [u"0", u"1"]
+    _age = [u"0", u"1"]
+    _gender = [u"M", u"W", u"U"]
+    _accent = [u"Y", u"N"]
+    _folder_audio = "clips"
+    sample_rate = 22050
+
+    @classmethod
+    def setUpClass(cls):
+        cls.root_dir = os.path.join(cls.get_base_temp_dir(), 'waves_commonvoice')
+        # The path convention commonvoice uses
+        base_dir = os.path.join(cls.root_dir, commonvoice.FOLDER_IN_ARCHIVE, commonvoice.VERSION, "en")
+        os.makedirs(base_dir, exist_ok=True)
+
+        # Tsv file name difference does not mean different subset, testing as a whole dataset here
+        tsv_filename = os.path.join(base_dir, commonvoice.DEFAULT_TSV)
+        with open(tsv_filename, "w", newline='') as tsv:
+            writer = csv.writer(tsv, delimiter='\t')
+            writer.writerow(cls._headers)
+            for i in range(100):
+                audio_filename = f'voice_{i:05d}.wav'
+
+                # Generate and store text data
+                text_data = [
+                    random.choice(cls._client_ids),
```

Comment:
Adding randomness to test is tricky, because in case of failure, you need to log everything that's required to reproduce the failure. In this case, random seed should do but we cannot really guarantee always that's the case.

Then the question is "Are we benefitting from adding randomness here?" or "Is randomness adding wider coverage of test cases?", and I do not think so, because after all, the generated samples are homogeneous in how they are processed by Dataset class. What matters more here is to have multiple of samples (which this code covers), because that way, we can see the ordering that Dataset traverses. (Which is the ultimate motivation to add this test. See https://github.com/pytorch/audio/issues/794 for some context.)

Instead I suggest to simply hard code some contents (preferably on module level in the hope for the improved readability)

```python
# on module level
train_csv_headers = ["client_ids", "path", "sentence", "up_votes", "down_votes", "age", "gender", "accent"]
# Note: extension is changed to wav for the sake of test
# Note: the first content is missing values for `age`, `gender` and `accent` as in the original data.
train_csv_contents = [
        ("9d16c5d980247861130e0480e2719f448be73d86a496c36d01a477cbdecd8cfd1399403d7a77bf458d211a70711b2da0845c1afdc5f1f694a2bbbccd162a34bf", "common_voice_en_18885784.wav", "He was accorded a State funeral, and was buried in Drayton and Toowoomba Cemetery.", "2", "0", "", "", "")
    ("c82eb9291328620f06025a1f8112b909099e447e485e99236cb87df008650250e79fea5ca772061fb6a370830847b9c44d2078cfa0f601b1f8cf6b98c479a7fc", "common_voice_en_556542.wav", "Once more into the breach", "2", "0", "thirties", "male", "us"),
    ("f74d880c5ad4c5917f314a604d3fc4805159d255796fb9f8defca35333ecc002bdf53dc463503c12674ea840b21b4a507b7cd0b3c636434f9a06bd81538de4f9", "common_voice_en_18607573.wav", "Caddy, show Miss Clare and Miss Summerson their rooms.", "2", "0", "twenties", "male", "canada"),
]

# in setUpClass
writer.writerow(train_csv_headers)
for i, content in enumerate(train_csv_contents):
    writer.write(content)
    audio_filename = content[1]

    data = get_whitenoise(sample_rate=cls.sample_rate, duration=1, n_channels=1, seed=i, dtype='float32')
```